src/components/cursor-controller.js

```diff
@@ -0,0 +1,393 @@
+    window.addEventListener("exit-vr", this.exitVRListener);
+
+    this.el.addEventListener("raycaster-intersection", this.raycasterIntersectionListener);
+    this.el.addEventListener("raycaster-intersection-cleared", this.raycasterIntersectionClearedListener);
+
+    this.data.playerRig.addEventListener("model-loaded", this.modelLoadedListener);
+
+    this.el.sceneEl.addEventListener("controllerconnected", this.controllerConnectedListener);
+    this.el.sceneEl.addEventListener("controllerdisconnected", this.controllerDisconnectedListener);
+  },
+
+  pause: function() {
+    document.removeEventListener("mousedown", this.mouseDownListener);
+    document.removeEventListener("mousemove", this.mouseMoveListener);
+    document.removeEventListener("mouseup", this.mouseUpListener);
+    document.removeEventListener("wheel", this.wheelListener);
+
+    window.removeEventListener("enter-vr", this.enterVRListener);
+    window.removeEventListener("exit-vr", this.exitVRListener);
+
+    this.el.removeEventListener("raycaster-intersection", this.raycasterIntersectionListener);
+    this.el.removeEventListener("raycaster-intersection-cleared", this.raycasterIntersectionClearedListener);
+
+    if (this.controller != null) {
+      this.controller.removeEventListener(this.data.controllerEvent, this.controllerEventListener);
+      this.controller.removeEventListener(this.data.controllerEndEvent, this.controllerEndEventListener);
+      this.controller.removeEventListener(this.data.grabEvent, this.controllerEventListener);
+      this.controller.removeEventListener(this.data.releaseEvent, this.controllerEndEventListener);
+    }
+
+    this.data.playerRig.removeEventListener("model-loaded", this.modelLoadedListener);
+
+    this.el.sceneEl.removeEventListener("controllerconnected", this.controllerConnectedListener);
+    this.el.sceneEl.removeEventListener("controllerdisconnected", this.controllerDisconnectedListener);
+  },
+
+  tick: (function() {
+    let wasIntersecting = false;
+    let wasPhysicalHandGrabbing = false;
+    const origin = new THREE.Vector3();
+    const direction = new THREE.Vector3();
+    const point = new THREE.Vector3();
+    const controllerQuaternion = new THREE.Quaternion();
+
+    return function() {
+      this.isGrabbing = this.data.cursor.components["super-hands"].state.has("grab-start");
+
+      //handle physical hand
+      if (this.physicalHand) {
+        const state = this.physicalHand.components["super-hands"].state;
+        const isPhysicalHandGrabbing = state.has("grab-start") || state.has("hover-start");
+        if (wasPhysicalHandGrabbing != isPhysicalHandGrabbing) {
+          this.data.cursor.setAttribute("visible", !isPhysicalHandGrabbing);
+          this.el.setAttribute("line", { visible: !isPhysicalHandGrabbing });
+          this.currentTargetType = TARGET_TYPE_NONE;
+        }
+        wasPhysicalHandGrabbing = isPhysicalHandGrabbing;
+        if (isPhysicalHandGrabbing) return;
```

Comment:
This early return looks a bit sneaky, I could imagine it burning us in the future if we end up adding something to tick that we want to always happen.